test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetRestoreCommandTest.cs

```diff
@@ -2737,6 +2737,81 @@ public void RestoreCommand_PackageNamespaceLongerMatches_Succeed()
+        {
+            // Arrange
+            var nugetexe = Util.GetNuGetExePath();
+
+            using (var workingPath = TestDirectory.Create())
+            {
+                var proj1Directory = Path.Combine(workingPath, "proj1");
+                Directory.CreateDirectory(proj1Directory);
+
+                var proj1File = Path.Combine(proj1Directory, "proj1.csproj");
+                File.WriteAllText(
+                    proj1File,
+                    @"<Project ToolsVersion='4.0' DefaultTargets='Build'
+    xmlns='http://schemas.microsoft.com/developer/msbuild/2003'>
+  <PropertyGroup>
+    <OutputType>Library</OutputType>
+    <OutputPath>out</OutputPath>
+    <TargetFrameworkVersion>v4.0</TargetFrameworkVersion>
+  </PropertyGroup>
+  <ItemGroup>
+    <None Include='packages.config' />
+  </ItemGroup>
+</Project>");
+
+                var opensourceRepositoryPath = Path.Combine(workingPath, "PublicRepository");
+                Directory.CreateDirectory(opensourceRepositoryPath);
+                Util.CreateTestPackage("My.MVC.ASP", "1.0.0", opensourceRepositoryPath);
+                Util.CreateTestPackage("Contoso.Opensource.Buffers", "1.0.0", opensourceRepositoryPath);
+
+                Util.CreateFile(proj1Directory, "packages.config",
+@"<packages>
+  <package id=""My.MVC.ASP"" version=""1.0.0"" targetFramework=""net461"" />
+</packages>");
+
+                var configPath = Path.Combine(workingPath, "nuget.config");
+                SettingsTestUtils.CreateConfigurationFile(configPath, $@"<?xml version=""1.0"" encoding=""utf-8""?>
+<configuration>
+    <packageSources>
+    <!--To inherit the global NuGet package sources remove the <clear/> line below -->
+    <clear />
+    <add key=""PublicRepository"" value=""{opensourceRepositoryPath}"" />
+    </packageSources>
+    <packageSourceMapping>
+        <packageSource key=""PublicRepository"">
+            <package pattern=""Contoso.MVC.ASP"" />   <!-- My.MVC.ASP doesn't match any package name spaces -->
+        </packageSource>
+    </packageSourceMapping>
+</configuration>");
+
+                var packagePath = Path.Combine(workingPath, "packages");
+
+                string[] args = new string[]
+                    {
+                        "restore",
+                        proj1File,
+                        "-solutionDir",
+                        workingPath,
+                        "-verbosity detailed"                    };
+
+                // Act
+                var r = CommandRunner.Run(
+                    nugetexe,
+                    workingPath,
+                    string.Join(" ", args),
+                    waitForExit: true);
+
+                // Assert
+                Assert.Equal(_failureCode, r.ExitCode);
+                Assert.Contains("Package namespace match not found for package ID 'My.MVC.ASP'", r.AllOutput);
+                Assert.Contains("NU1000: Unable to find version '1.0.0' of package 'My.MVC.ASP'.", r.Item3);
```

Comment:
In package reference scenarios, https://docs.microsoft.com/en-us/nuget/reference/errors-and-warnings/nu1102 error code is used if a specific version of a package is not found in the configured sources. I am not sure if `NU1000` is correct in this scenario. Is an exception thrown in PC scenarios if a package is not found instead of logging it using NU110* error codes?

Reply:
It's not clear from [this comment](https://github.com/NuGet/NuGet.Client/pull/4248#discussion_r703670430), but the code changed is a generic `Exception` handler, not a specific error handler. In order to correctly report NU1102 in this scenario, the relevant code needs to log the NU error at that location, not throw an exception. In my opinion, it's out of scope for this issue to fix that.

Reply:
I agree Andy. It is out of scope for this issue but want to raise awareness that `NU1000` code may not be relevant here as it does not match with package reference scenarios.

Reply:
The point of NU1000 is that it's an error that hasn't been given a specific error code. Ideally NU1000 should not be used anywhere, ever. But we have this generic exception handler where we're not sure what the specific error actually is.